wallet/wallet.h

```diff
@@ -363,6 +363,8 @@ struct invoice {
 	u64 *msatoshi;
 	/* Set if state == PAID */
 	u64 msatoshi_received;
+	/* Set if state == PAID */
+	u64 paid_timestamp;
```

Comment:
Can't find it being set anywhere before `state == PAID`, should probably be initialized to `0` [here](https://github.com/ZmnSCPxj/lightning/blob/51ba88213fe2cd67ee20dcb6b1e49b9d7412d878/wallet/invoices.c#L184) just to mimic loading from DB as well.

Reply:
`msatoshi_received` is similar also (i.e. not set anywhere before `state == PAID`).  indeed `pay_index` and `msatoshi_received` are not handled similarly (`pay_index` gets set to 0 always, `msatoshi_received` is only set if `state == PAID`), so we should reconcile those two first.

Reply:
In theory, not loading the fields should help valgrind detect cases where the program erroneously reads those fields when `state != PAID`.  So I suppose I can understand some rationale for not loading a field that has no valid value at  this point anyway.

Reply:
Yes, I prefer non-initialization for exactly this reason, even when loading from db.

I pondered adding an "uninitialized(x)" macro which would memset to 0 (or 0x42424242...) iff not running under valgrind (which is *usually* less disasterous than leaking some random value).

Reply:
Okay, I will add a commit to *not* set `pay_index` if not PAID instead.

Reply:
Hmm, I'm going to break this with the JSON API patches :(

But you'll also need to add the field to doc/lightning-listinvoices.7.txt after that...

Reply:
Maybe just cherry-pick this particular commit and add it to your JSON API pull request?  Probably after removing the unnecessary "CREATE INDEX invoices_paid_timestamp".  Then I'll just put the "not set `pay_index` if not PAID" in a separate pull request.